site/app/templates/admin/admin_gradeable/AdminGradeablePeers.twig

```diff
@@ -169,8 +183,134 @@
+        });
+        let new_grader = !existing_graders.includes(user_input.val());
+        // if both above conditions are met, allow user to submit
+        return (valid_entry && new_grader);
+    }
+
+    function updateSubmitButton(valid_grader, valid_students, button_id, new_grader) {
+        // check that the grader is valid, all new students entered are valid, and 
+        // that there's at least one new student added if adding new grader
+        if (valid_grader && !valid_students.includes(false) && (!new_grader || valid_students.includes(true))) {
+            $(button_id).prop('disabled', false);
+        } else {
+            $(button_id).prop('disabled', true);
+        }
+    }
+
+    function submitFormAjax2() {
+        let url = $('#peer_grader_submit_url').val();
+        let form = $("#admin-gradeable-peers-form2");
+        let fd = new FormData();
+        let csrf_token = $("#csrf_token2", form).val();
+        fd.append('csrf_token', csrf_token);
+        let new_grader_id = $(":text#new_peer_grader", form).val();
+        fd.append('new_grader_id', new_grader_id);
+        // set to check for duplicate user entries
+        let add_student_ids = new Set();
+        $(".add_user_ids", form).each(function() {
+            if ($(this).val() != "") add_student_ids.add($(this).val());
+        });
+        let add_student_ids_arr = JSON.stringify(Array.from(add_student_ids));
+        fd.append('add_student_ids', add_student_ids_arr);
+        $.ajax({
+            url: url,
+            type: 'POST',
+            data: fd,
+            processData: false,
+            contentType: false,
+            success: function(data) {
+                $('.popup-form').css('display', 'none');
+                let new_peers = JSON.parse(data)["data"];
+                // temporarily add new grader's row to the end of the peer table
+                let new_row_html = '<tr><td class="peer-table-grader">'+ new_grader_id +'</td><td class="peer-table-students">';
+                let new_students = new_peers[new_grader_id];
+                for (i in new_students) {
+                    new_row_html += '<span>'+new_students[i]+'</span> ';
+                }
+                new_row_html += '</td><td><a aria-label="edit'+new_grader_id+'\'s students"><i class="fas fa-pencil-alt"></i></a></td>';
+                $('#peer-table tr:last').after(new_row_html);
+                // move the row to where it belongs lexicographically
+                let new_grader_index = Object.keys(new_peers).indexOf(new_grader_id) + 1;
+                // if it already belongs in the last row, then leave it
+                if (new_grader_index != $('#peer-table tr').length - 1) {
+                    let next_tr = $('#peer-table tr:eq(' + new_grader_index + ')');
+                    $('#peer-table tr:last').insertBefore(next_tr);
+                }
+                
+            },
+            error: function() {
```

Comment:
lets have the error function accept a parameter and log it in case there's relevent information.
```
error function (err) {
   console.error(err);